docs/guides/main.rst

```diff
@@ -125,6 +125,25 @@ trouble or you have more detail to contribute here, please direct it to
 .. _install pip: http://www.pip-installer.org/en/latest/installing.html#install-pip
 .. _get-pip.py: https://raw.github.com/pypa/pip/master/contrib/get-pip.py
 
+Beets Workflow
+--------------
+
+In this section we introduce a briefly description of the beets workflows, which might help you to 
+understand the basics behind the program and the key commands. In case that you want
+to contribute as a programmer, make sense about how to navigate through the beets
+components and give better structure to your propose future plugins.
+
+* **Data Model**. The beets *Library* is a database that catalogs your music, which does not contain any music files; as a database, it saves the local links to your music files. The *Library* is composed by *Items* (songs) and *Albums*. The Album is a sequence of *Items* and consist of a set of album-level metadata that is share among the *Items*. Each *Item* belongs to an *Album*, songs without *Album* are *singleton*. The command to import music to your *Library* is ``beet import``, which consist of adding *Items* to the *Library* corresponding to files that exist on your disk. It is important to keep in mind that the importer perform different tasks at the same time such as copy, renaming, cleaning up metadata etc. There is an option to disable this multitask function by using ``beet import -AC``. The *Library* is essential because *everything* beets does relies on this component.
```

Comment:
Should the discussion about `beet import` really go here? We're about to introduce the commands, but this section (if I understand correctly) is about concepts more than about specific commands.